tiledb/sm/filesystem/s3.cc

```diff
@@ -947,35 +976,50 @@ Status S3::init_client() const {
   }
 #endif
 
+  if (!aws_access_key_id.empty() && !aws_secret_access_key.empty() &&
```

Comment:
One last thing: instead of having this if-statement, put an `} else {` on L1009 just so that we don't have duplicate logic checking for the the different credential types. So:
```
if (!aws_access_key_id.empty() && !aws_secret_access_key.empty()) {
 ...
} else if (!aws_role_arn.empty()) {
 ...
} else {
  return Status::S3Error(
        "Ambiguous authentication credentials; both permanent and temporary "
        "authentication credentials are configured");
}
```

Reply:
True!! [here](https://github.com/TileDB-Inc/TileDB/pull/1882#discussion_r514382857)